tiling.js

```diff
@@ -742,12 +742,12 @@ var Space = class Space extends Array {
         if (actor)
             actor.remove_clip();
 
-        this.layout();
         if (selected) {
             ensureViewport(selected, this);
         } else {
             this.selectedWindow = null;
         }
+        this.layout();
```

Comment:
Hmmm, I was thinking of this change while driving today.  Although moving layout to the end for this use case might solve that issue - it might cause issues for other use cases.

Basically, the normal approach is "layout to configure all the tiled windows" and then "ensureViewport to bring the selected window into view", which also sets the tracked `seletectedWindows` variable.

Anyways, what I'm trying to say is that instead of changing the layout() position to be in an unnatural position (e.g. ensuring viewport then layout doesn't really make sense, and the layout may change window positions etc.), it would seem better to set `this.selectedWindow` to `selected` if it exists.

Will test this out a bit.  Although I'm having trouble reproducing the error in the first place (just checking, are you on X11 or Wayland?).

Reply:
Will test this approach a bit and compare against just moving layout().

Reply:
P.S. I may be overthinking things...